plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -266,10 +266,11 @@ public Answer copyTemplateToPrimaryStorage(final CopyCommand cmd) {
 
                 Map<String, String> details = primaryStore.getDetails();
 
-                String path = details != null ? details.get("managedStoreTarget") : null;
+                String path = derivePath(primaryStore, destData, details);
 
                 if (!storagePoolMgr.connectPhysicalDisk(primaryStore.getPoolType(), primaryStore.getUuid(), path, details)) {
                     s_logger.warn("Failed to connect physical disk at path: " + path + ", in storage pool id: " + primaryStore.getUuid());
+                    return new PrimaryStorageDownloadAnswer("Failed to spool template disk at path: " + path + ", in storage pool id: " + primaryStore.getUuid());
```

Comment:
I tested this branch now and this line breaks the Linstor driver on a fresh installation.

AFAIK: on the first template (systemtemplate) copy ACS doesn't provide a size for the destination template and
so Linstor ignores the copy request and also the agent issues a `.connectPhysicalDisk()` with `null` as volumePath and the LinstorAdaptor ignores this. 
Later ACS does the correct thing and copies the template from the image store to the Linstor primary storage.

BUT this Failure answer breaks this code.

Reply:
This has been removed, not returning it appears to have no negative consequences on the Primera and Pure driver.